src/node_api.rs

```diff
@@ -139,6 +143,27 @@ impl Node {
 		}
 	}
 
+	// build an offer to receive payments at this node. path_pubkeys lists the nodes the path will contain,
+	// starting with the introduction node and ending in the destination node (the current node).
+	pub async fn create_offer(
+		&self, path_pubkeys: &[PublicKey], network: Network, msats: u64, quantity: Quantity,
+	) -> Result<Offer, Bolt12SemanticError> {
+		let secp_ctx = Secp256k1::new();
+		let path =
+			BlindedPath::new_for_message(path_pubkeys, &*self.keys_manager, &secp_ctx).unwrap();
+		let (pubkey, _) = self.get_node_info();
+
+		let expiration = SystemTime::now() + Duration::from_secs(24 * 60 * 60);
```

Comment:
_Might_ want to make this configurable? Can see us wanting to test expiry in the future?